src/login/components/LoginFormWalletFile/LoginFormWalletFile.js

```diff
@@ -47,22 +52,59 @@ export default class LoginFormWalletFile extends React.Component {
   }
 
   renderAccounts = () => {
-    const { accounts, wif } = this.props;
+    const { accounts, encryptedWIF } = this.props;
 
     if (accounts.length === 0) {
       return null;
     }
 
     return (
-      <Select className={styles.accounts} value={wif} onChange={this.handleSelect}>
-        <option value="">Select an account</option>
-        {map(this.props.accounts, (account, index) => (
-          <option value={account.encrypted} key={`account${index}`}>{account.label}</option>
-        ))}
-      </Select>
+      <div>
+        <Select className={styles.accounts} value={encryptedWIF} onChange={this.handleSelect}>
+          <option value="">Select an account</option>
+          {map(this.props.accounts, (account, index) => (
+            <option value={account.encrypted} key={`account${index}`}>{account.label}</option>
+          ))}
+        </Select>
+        {this.renderPassphraseInput()}
+        {this.renderDescription()}
+      </div>
     );
   }
 
+  renderPassphraseInput = () => {
+    const { encryptedWIF, passphrase } = this.props;
+
+    let input;
+    if (encryptedWIF !== '' && !wallet.isPrivateKey(encryptedWIF)) {
```

Comment:
💅 Can we drop the `let` and inverse the logic here to simply return early if needed?  i.e.:

```js
if (encryptedWIF === '' || wallet.isPrivateKey(encryptedWIF)) {
  return null;
}

return (
  <Input ... />
);
```